client/js/lounge.js

```diff
@@ -469,21 +469,21 @@ $(function() {
 			.find("#chan-" + data.chan)
 			.find(".messages");
 
+		// get the scrollable wrapper around messages
+		var scrollable = chan.closest(".chat");
+		var heightOld = chan.height();
+		chan.prepend(documentFragment).end();
+
 		// Remove the date-change marker we put at the top, because it may
 		// not actually be a date change now
 		var children = $(chan).children();
-		if (children.eq(0).hasClass("date-marker")) { // Check top most child
+		if (children.eq(0).hasClass("date-marker-container")) { // Check top most child
 			children.eq(0).remove();
-		} else if (children.eq(0).hasClass("unread-marker") && children.eq(1).hasClass("date-marker")) {
-			// Otherwise the date-marker would get 'stuck' because of the new-message marker
+		} else if (children.eq(0).hasClass("unread-marker") && children.eq(1).hasClass("date-marker-container")) {
```

Comment:
Hmm. I guess it isn't needed, because there will never be normal messages at index 0